app/battle/knight.py

```diff
@@ -0,0 +1,44 @@
+class Knight:
+    def __init__(self,
+                 name: str,
+                 power: int,
+                 hp: int,
+                 armour: list,

```

Comment:
Improve type annotations across the project to indicate a list of what instances you expect to obtain for different arguments